### ISSUE TITLE
[Doc]: Wrong source hyperlink for plt.grid

### ISSUE THREAD

**Issue by StFroese** [matplotlib/matplotlib#22629]
### Documentation Link

https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.grid.html

### Problem

* the [[source]](https://github.com/matplotlib/matplotlib/blob/v3.5.1/lib/matplotlib/_api/deprecation.py#L2559-L2561) link in the documentation leads to the deprecation.py and not to the pyplot.py

### Suggested improvement

* the link to the the correct one which leads to pyplot.py [[source]](https://github.com/matplotlib/matplotlib/blob/v3.5.1/lib/matplotlib/pyplot.py#L2559-L2561)

**Comment by timhoffm** [comment]
Same for the `Axes.grid()` mehod:
https://matplotlib.org/stable/api/_as_gen/matplotlib.axes.Axes.grid.html?highlight=grid#matplotlib.axes.Axes.grid

So it's likely the `@rename_parameter` decorator that needs fixing.

**Comment by StFroese** [comment]
@timhoffm do you know the location where this could be fixed in the code or where I should take a look?

**Comment by oscargus** [comment]
Here is the rename_parameter decorator that Tim suspected.

https://github.com/matplotlib/matplotlib/blob/945692c900b23f4c018b00acb362193376f3a7f9/lib/matplotlib/_api/deprecation.py#L257-L305

**Comment by timhoffm** [comment]
@StFroese It's likely an interplay between code introspection (used in sphinx) and the decorator. I don't know your background, but I recommend that you should have reasonable knowlege how these work in Python if you want to look into the issue. Otherwise it's probably very hard to understand and fix it.

**Comment by StFroese** [comment]
@timhoffm yeah I think this is a little bit over my level. I'm not very familiar with how introspection exactly works. So this might be an issue for a more experienced person.

**Comment by timhoffm** [comment]
The specific issue does not exist anymore because the deprecation has expired. I don’t think it’s worth digging into the details.
